lib/simplifyExpression/basicsSearch/removeAdditionOfZero.js

```diff
@@ -1,30 +1,30 @@
-const clone = require('../../util/clone');
+const {rewriteNode} = require('math-rules');
 
 const ChangeTypes = require('../../ChangeTypes');
+const defineRuleString = require('../../util/defineRuleString');
 const Node = require('../../node');
 
+// e.g. 2 + 0 -> 2
+const REMOVE_ADDING_ZERO = defineRuleString('#a + 0', '#a');
+
+// e.g. 0 + 2 -> 2
+const REMOVE_ADDING_ZERO_REVERSE = defineRuleString('0 + #a', '#a');
+
```

Comment:
I'll add a comment.  `math-rules` treats addition and multiplication in a special way.  Since an expression like `a + b + 0 + d + e` is a single `add` node, we'd don't want to have to describe rules for all of the different possible sums that could be expressed.  When `math-rules` is matching an `add` or `mul` pattern it will compare each sub-array of node.args against the array of args in the pattern.  It will compare `0 + #a` against `a + b`, `b + 0`, `0 + d`, etc. until it finds a match.  In this case it finds `0 + d` which is replaces with `d`.